.eslintrc

```diff
@@ -7,18 +7,36 @@
   ],
   "overrides": [
     {
-      "files": ["**/*.ts?(x)"],
-      "plugins": ["@typescript-eslint"],
+      "files": [
```

Comment:
I see that some unrelated lines tot changed? Is that maybe your IDE? Or something applying some custom format?

Reply:
I'll check on that I am using `neovim` and may my `json-lsp` is not aligned with that what are you guys using to format this ?

Reply:
Figured this one out, turns out you guys are using `prettier` for formatting you `json`

Reply:
Sure, but I’ve edited this file countless times. Why it only got reformatted by you? 🙃

Reply:
To be more precise here, the “.rc” files (or at least a few of them) are indeed JSON files but they don’t have a JSON file extension.

So it is erroneous for prettier to simply format it out of the blue (except if you have prettier enabled for that file)

Our pre-commit hook definitely will not format those files, and that could theoretically be “solved” if we applied JSON extensions to these files (cc @bmuenzenmeyer, as I sadly think we might want to do that sooner than GHC)

Since the file is not being linted/formatted by precommit hooks and has no apparent file extension, its formatting will clearly depend on how your IDE decides to format that file.

That can be a combination of extensions and sort of IDE you have.

Reply:
TL;DR, this file shouldn’t have been formatted, and yes, we are aware that this flakiness of formatting is due to the file not being actually handled by precommit rules and there are other things at play, there’s an open issue for that.

But it is honestly too much draining from my side to explain every little detail for a newcomer on their first PR. Hence **why** I’m simply asking you to follow my recommendations, as I would except you to expect the maintainer of an open source project to know (to a certain extent) what they’re talking about 😅

Reply:
its just I had a different plugin/extension which was taken precedence over prettier

Reply:
Sorry for the big texts @umairraza96 just trying to give you some context here 🙇🙏

Reply:
I don't mind at all, I deeply appreciate the level of details you shared with me :)

Reply:
> its just I had a different plugin/extension which was taken precedence over prettier

Yeah, that makes sense, and `lint-staged` would be fixing that, **if** the `.rc` files had proper file extensions.